packages/cds-odata-annotation-converter/src/transforms/annotation/handlers/annotation.ts

```diff
@@ -1,20 +1,28 @@
 import type { Annotation, AnnotationNode, Identifier } from '@sap-ux/cds-annotation-parser';
-import { copyRange, ANNOTATION_TYPE, nodeRange, EMPTY_VALUE_TYPE } from '@sap-ux/cds-annotation-parser';
+import {
+    copyRange,
+    ANNOTATION_TYPE,
+    nodeRange,
+    EMPTY_VALUE_TYPE,
+    ReservedProperties,
+    STRING_LITERAL_TYPE
+} from '@sap-ux/cds-annotation-parser';
 
 import type { Element } from '@sap-ux/odata-annotation-core-types';
-import { Range, createElementNode, Edm, Position } from '@sap-ux/odata-annotation-core-types';
+import { Range, createElementNode, Edm, Position, createAttributeNode } from '@sap-ux/odata-annotation-core-types';
 import { convertFlattenedPath } from '../flattened';
 
 import type { ConvertResult, NodeHandler, Subtree } from '../handler';
 import { getTerm } from '../type-resolver';
 import type { VisitorState } from '../visitor-state';
 import { createQualifierAttribute, createTermAttribute } from '../creators';
+import { createRange } from '../range';
 
 export const annotationHandler: NodeHandler<Annotation> = {
     type: ANNOTATION_TYPE,
     convert,
     getChildren(state: VisitorState, annotation: Annotation): AnnotationNode[] {
-        if (annotation.value) {
+        if (annotation.value && !state.context.isFlattened$Type) {
```

Comment:
This logic should not relay on context, but annotation value itself. Context is something that is coming from from nodes higher up in the tree. If we want to avoid recomputing some data between `convert` and `getChildren` calls we could think of some other mechanism. However, currently I don't think checking if the the annotation term contains $Type again would be expensive.

Reply:
adapted